### PR TITLE
guest_os_booting: new case of booting vm with seclabel in backed nvram

### PR DESCRIPTION
**Pull request by meinaLi**
This case is to verify the seclabel source in backed nvram: VIRT-299960

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.ovmf_seclabel_in_nvram
JOB ID     : f2363d07bafc67fb4bd78af09d7d22580d32a42b
JOB LOG    : /var/lib/avocado/job-results/job-2023-11-08T03.38-f2363d0/job.log
 (1/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.positive_test.with_selinux_seclabel: PASS (37.54 s)
 (2/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.positive_test.with_dac_seclabel: PASS (35.10 s)
 (3/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.negative_test.invalid_dac_seclabel: PASS (5.30 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-08T03.38-f2363d0/results.html
JOB TIME   : 78.69 s
```

**Comment by Yingshun** [comment]
@yafu-1 @zhentang-tz The manual case was not reviewed by feature owners, could you please help review the manual one first? Thanks!

**Review by chunfuwen — APPROVED** [review]
lgtm

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_seclabel_in_nvram.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    vm = env.get_vm(vm_name)
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    bkxml = vmxml.copy()
+
+    try:
+        os_attrs = {**os_dict, **nvram_attrs, **nvram_source}
+        guest_os.prepare_os_xml(vm_name, os_attrs, firmware_type)
+        vmxml = guest_os.check_vm_startup(vm, vm_name, error_msg)
+        if error_msg:
```
Comment (yafu-1) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_seclabel_in_nvram.py:60]: @meinaLi  Does the error_msg print in the log file here?

**Comment by yafu-1** [comment]
@meinaLi It's better to also add relabel='no' in this testcase.

**Comment by Yingshun** [comment]
> @meinaLi It's better to also add relabel='no' in this testcase.

+1. And it's better to cover different label values I think.(Plz refer to VIRT-296951/296954).

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.ovmf_seclabel_in_nvram
JOB ID     : 727c92917065641863b6864779fc46e28ee65fc5
JOB LOG    : /var/lib/avocado/job-results/job-2023-12-07T03.46-727c929/job.log
 (1/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.positive_test.with_selinux_seclabel: PASS (37.81 s)
 (2/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.positive_test.with_dac_seclabel: PASS (35.80 s)
 (3/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.negative_test.invalid_selinux_seclabel: PASS (6.04 s)
 (4/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.negative_test.invalid_dac_seclabel: PASS (5.22 s)
 (5/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_seclabel_in_nvram.negative_test.with_no_relabel: PASS (5.15 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-12-07T03.46-727c929/results.html
JOB TIME   : 90.83 s
```

**Comment by meinaLi** [comment]
Added another negative test scenarios in this case.

**Review by yafu-1 — APPROVED** [review]
LGTM

**Review by Yingshun — APPROVED** [review]

**Pull request merged**